src/main/js/helpers/axis.js

```diff
@@ -205,26 +221,67 @@ const getAxesScale = (axis, scale, config) => {
             config.axis.y2.domain.upperLimit - config.axis.y2.domain.lowerLimit
         );
 
+    if (
+        config.suppressTrailingZeros &&
+        utils.isUndefined(config.axis.y.ticks.format)
+    ) {
+        axis.y = d3.axisLeft(scale.y);
+        tickFormatToTrimTrailingZeros = tickFormatter(axis.y);
+    }
+
     axis.y = prepareYAxis(
         scale.y,
         config.axis.y.domain,
         config.height,
-        getAxisTickFormat(config.locale, config.axis.y.ticks.format),
+        getAxisTickFormat(
+            config.locale,
+            utils.isDefined(config.axis.y.ticks.format)
+                ? config.axis.y.ticks.format
+                : tickFormatToTrimTrailingZeros
+        ),
         config.ticksCount
     );
 
     if (hasY2Axis(config.axis)) {
+        if (
+            config.suppressTrailingZeros &&
+            utils.isUndefined(config.axis.y2.ticks.format)
```

Comment:
Can you add a comment here as well?

Reply:
Added comment [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/cd62efa8c9feea2ab4a5585cb63fa492169425ae).